gax-java/gax/src/main/java/com/google/api/gax/core/GaxProperties.java

```diff
@@ -148,4 +148,30 @@ static Optional<String> getBundleVersion(Class<?> clazz) {
       return Optional.empty();
     }
   }
+
+  /**
+   * Returns the Protobuf runtime version as reported by com.google.protobuf.RuntimeVersion, if
+   * class is available, otherwise by reading from MANIFEST file. If niether option is available
+   * defaults to protobuf version 3 as RuntimeVersion class is available in protobuf version 4+
+   */
+  @VisibleForTesting
+  static String getProtobufVersion(ClassWrapper classWrapper, Class clazz) {
+    try {
+      Class<?> protobufRuntimeVersionClass =
+          classWrapper.forName("com.google.protobuf.RuntimeVersion");
+      return classWrapper.getFieldValue(protobufRuntimeVersionClass, "MAJOR")
+          + "."
+          + classWrapper.getFieldValue(protobufRuntimeVersionClass, "MINOR")
+          + "."
+          + classWrapper.getFieldValue(protobufRuntimeVersionClass, "PATCH");
+    } catch (ClassNotFoundException
+        | NoSuchFieldException
+        | IllegalAccessException
+        | SecurityException e) {
+      // If manifest file is not available default to protobuf generic version 3 as we know
+      // RuntimeVersion class is
+      // available in protobuf jar 4+.
+      return getBundleVersion(clazz).orElse("3");
```

Comment:
So we are going to have two different format of protobuf versions (3.x.x vs 3)? Is it going to cause issues when we try to aggregate the metrics?

Reply:
There might be 3.1.2, which provides precision. But there shouldn't be any 3.x.x.

We shouldn't be providing precision (significant digits) when we don't know them. We can fix any parsing logic as necessary.